packages/cardano-services/src/WsServer/server.ts

```diff
@@ -0,0 +1,278 @@
+// cSpell:ignore njson
+
+import { Cardano, WSMessage } from '@cardano-sdk/core';
+import { HandleRequestOptions, handleRequest } from './requests';
+import { Logger } from 'ts-log';
+import { NJSON } from 'next-json';
+import { Notification, Pool } from 'pg';
+import { Server, createServer } from 'http';
+import { WebSocket, WebSocketServer } from 'ws';
+import { initDB } from './db';
+import { v4 } from 'uuid';
+
+export interface WsServerConfiguration {
+  /** The heartbeat timeout in seconds. */
+  heartbeatTimeout?: number;
+
+  /** The port to listen. */
+  port: number;
+}
+
+export interface WsServerDependencies extends HandleRequestOptions {
+  /** The logger. */
+  logger: Logger;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = () => {};
+
+export class CardanoWsServer {
+  private closeNotify: () => void;
+  private closing = false;
+  private db: Pool;
+  private logger: Logger;
+  private notifyConnected = false;
+  private server: Server;
+  private tip: Cardano.Tip;
+  private wss: WebSocketServer;
+
+  constructor(dependencies: WsServerDependencies, cfg: WsServerConfiguration) {
+    this.db = dependencies.db;
+    this.logger = dependencies.logger;
+
+    // Create the HTTP and the WebSocket servers
+    this.server = this.createHttpServer();
+    this.wss = new WebSocketServer({ server: this.server });
+
+    // Attach the handlers to the servers events
+
+    this.wss.on('connection', this.createOnConnection(dependencies, (cfg.heartbeatTimeout || 60) * 1000));
```

Comment:
Tiny nitpick: you could use `Seconds.toMilliseconds` for the transformation

Reply:
70f3fe7bc1 should solve this